docs/src/main/sphinx/admin/properties-client-protocol.md

```diff
@@ -0,0 +1,228 @@
+  HTTP requests per data segment.
+* `COORDINATOR_PROXY` - client accesses the coordinator and gets data segment
+  through it. Uses one client HTTP request per data segment, but requires a
+  coordinator HTTP request to the storage.
+* `WORKER_PROXY` - client accesses the coordinator, which redirects to an
+  available worker node. It  fetches the data from the storage and provides it
+  to the client. Uses two client HTTP requests, and requires a worker request to
+  the storage.
+
+### `protocol.spooling.encoding.json.enabled`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `true`
+
+Activate support for using JSON encoding for spooling segments.
+
+### `protocol.spooling.encoding.json+zstd.enabled`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `true`
+
+Activate support for using JSON encoding with Zstandard compression for spooling
+segments.
+
+### `protocol.spooling.encoding.json+lz4.enabled`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `true`
+
+Activate support for using JSON encoding with LZ4 compression for spooling
+segments.
+
+### `protocol.spooling.initial-segment-size`
+
+- **Type:** [](prop-type-data-size)
+- **Default value:** 8MB
+
+Initial size of the spooled segments.
+
+### `protocol.spooling.maximum-segment-size`
+
+- **Type:** [](prop-type-data-size)
+- **Default value:** 16MB
+
+Maximum size per segment.
+
+### `protocol.spooling.inlining.enabled`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `true`
+
+Allow spooled protocol to inline initial rows to decrease head latency.
+
+### `protocol.spooling.inlining.max-rows`
+
+- **Type:** [](prop-type-integer)
+- **Default value:** 1000
+
+Maximum number of rows to inline per worker.
+
+### `protocol.spooling.inlining.max-size`
+
+- **Type:** [](prop-type-data-size)
+- **Default value:** 128kB
+
+Maximum size of rows to inline per worker.
+
+(prop-spooling-file-system)=
+## Spooling file system
+
+The following properties are used to configure the object storage used with the
+[](protocol-spooling).
+
+### `fs.azure.enabled`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `false`
+
+Activate [](/object-storage/file-system-azure) for spooling segments. 
+
+### `fs.s3.enabled`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `false`
+
+Activate [](/object-storage/file-system-s3) for spooling segments. 
+
+### `fs.gcs.enabled`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `false`
+
+Activate [](/object-storage/file-system-gcs) for spooling segments. 
+
+### `fs.location`
+
+- **Type:** [](prop-type-string)
+
+The object storage location to use for spooling segments. Must be accessible by
+the coordinator and all workers. With the `protocol.spooling.retrieval-mode`
+retrieval modes `STORAGE` and `COORDINATOR_STORAGE_REDIRECT` the location must
+also be accessible by all clients. Valid location values vary by object storage
+type, and typically follow a pattern of `scheme://host:port/path`.
```

Comment:
No. Endpoint follows that for S3.

It's `scheme://bucketName/path/` like `s3://my-spooling-bucket/my-segments/`